File: app/src/main/java/com/bass9030/beabot/Replier.java
package com.bass9030.beabot;

import android.app.Notification;

public class Replier {
    private String packageName;
    private String room;
    private Notification.Action session;
    private boolean isDebugChat;
    private messageSender msgSender;

    public Replier(String packageName, String room, Notification.Action session, boolean isDebugChat) {
        this.packageName = packageName;
        this.room = room;
        this.session = session;
        this.isDebugChat = isDebugChat;
        this.msgSender = new messageSender();
    }

    public boolean reply(String msg) {
        return msgSender.send(this.session, msg, this.isDebugChat);
    }

    public boolean reply(String room, String msg) {
        return msgSender.send(room, this.packageName, msg, this.isDebugChat);
    }

    public boolean reply(String room, String msg, String packageName) {
        return msgSender.send(room, packageName, msg, this.isDebugChat);
    }

    public boolean replyDelayed(String msg, long ms) {
        return this.replyDelayed(this.room, msg, ms);
    }

    public boolean replyDelayed(String room, String msg, long ms) {
        try {
            Thread.sleep(ms);
            return msgSender.send(room, msg, this.isDebugChat);
        }catch(InterruptedException ex){ return false; }
    }

    public boolean markAsRead() {
        return this.markAsRead(this.room, this.packageName);
    }

    public boolean markAsRead(String room) {
        return msgSender.markAsRead(room, this.isDebugChat);
    }

    public boolean markAsRead(String room, String packageName) {
        return msgSender.markAsRead(room, packageName, this.isDebugChat);
    }
}

File: app/src/main/assets/Replier.js
const ReplierStr = `export declare module Replier {
    
    /**
     * 채팅이 수신된 채팅방으로 응답 전송 
     * @param {string} msg 
     * @returns {boolean}
     */
    function reply(msg: string): boolean;

    /**
     * 해당 채팅방으로 응답 전송 
     * @param {string} room 
     * @param {string} msg 
     * @returns {boolean}
     */
    function reply(room: string, msg: string): boolean;

    /**
     * 해당 메신저 앱의 해당 채팅방으로 응답 전송
     * @param {string} room 
     * @param {string} msg 
     * @param {string} packageName 
     * @returns {boolean}
     */
    function reply(room: string, msg: string, packageName: string): boolean;

    /**
     * ms 밀리초 뒤에 채팅이 수신된 채팅방으로 응답 전송 
     * @param {string} msg 
     * @param {number} ms 
     * @returns {boolean}
     */
    function replyDelayed(msg: string, ms: string): boolean;

    /**
     * ms 밀리초 뒤에 해당 채팅방으로 응답 전송 
     * @param {string} room 
     * @param {string} msg 
     * @param {boolean} ms 
     * @returns 
     */
    function replyDelayed(room: string, msg: string, ms: number): boolean;

    /**
     * 채팅이 수신된 채팅방에 응답을 보내지 않고 읽음으로 처리
     * @returns {boolean}
     */
    function markAsRead(): boolean

    /**
     * 해당 채팅방에 응답을 보내지 않고 읽음으로 처리
     * @param {string} room 
     * @returns {boolean}
     */
    function markAsRead(room: string): boolean;

    /**
     * 해당 앱의 해당 채팅방에 응답을 보내지 않고 읽음으로 처리
     * @param {string} room 
     * @param {string} packageName 
     * @returns {boolean}
     */
    function markAsRead(room: string, packageName: string): boolean;
}

declare global {
    var replierType: typeof Replier;
}`;